giturlparse/platforms/gitlab.py

```diff
@@ -4,9 +4,9 @@
 
 class GitLabPlatform(BasePlatform):
     PATTERNS = {
-        'https': r'https://(?P<domain>.+)/(?P<owner>.+)/(?P<repo>.+).git',
-        'ssh': r'git@(?P<domain>.+):(?P<owner>.+)/(?P<repo>.+).git',
-        'git': r'git://(?P<domain>.+)/(?P<owner>.+)/(?P<repo>.+).git',
+        'https': r'https://(?P<domain>.+)/(?P<owner>.+)/(?P<repo>.+)(?:\.git)?',
+        'ssh': r'git@(?P<domain>.+):(?P<owner>.+)/(?P<repo>.+)(?:\.git)?',
+        'git': r'git://(?P<domain>.+)/(?P<owner>.+)/(?P<repo>.+)(?\.git)?',
```

Comment:
I screwed this one up, should have been `?:` rather than just `?` (:angry: → self)